activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -212,6 +212,8 @@ def destroy(*records)
       def size
         if !find_target? || loaded?
           target.size
+        elsif @association_ids && target.empty?
+          @association_ids.length
```

Comment:
I prefer `@association_ids.size` since it is an only place using `length` in the class.